docs/enhancements/ove-ignition-generation.md

```diff
@@ -0,0 +1,165 @@
+2. Retrieve the cluster from the database to get the exact OpenShift version
+3. Create temporary directory structure with required assets:
+   - `cluster-manifests/infraenv.yaml` - InfraEnv manifest with user's SSH key
+   - `cluster-manifests/pull-secret.yaml` - User's pull secret for registry access
+   - `mirror/registries.conf` - Openshift Appliance specific registry mirror configuration
+4. Fetch the appropriate openshift-install binary matching the cluster version
+5. Execute `openshift-install agent create unconfigured-ignition --interactive`
+6. Return the ignition content
+
+**Key Constraints:**
+- InfraEnv must be bound to a cluster to determine the OpenShift version
+- The cluster's OpenShift version is used to ensure exact version match
+- The openshift-install binary version must match the OVE ISO version exactly
+- Registry configuration includes hardcoded OVE mirror registries:
+  - `quay.io/openshift-release-dev/ocp-release` → `registry.appliance.openshift.com:5000/openshift/release-images`
+  - `quay.io/openshift-release-dev/ocp-v4.0-art-dev` → `registry.appliance.openshift.com:5000/openshift/release`
+
+**Integration Points:**
+- `bareMetalInventory.V2DownloadInfraEnvFiles`: Modified to detect `disconnected-iso` type and call OVE ignition generation
+- Installer cache: Reused for managing openshift-install binaries
+- InfraEnv and Cluster models: Used to retrieve configuration data
+- Temporary directory management: Creates isolated workspace for each generation
+
+### Risks and Mitigations
+
+**Risk**: Version mismatch between OVE ISO and generated ignition
+- **Mitigation**: Use cluster's exact OpenShift version when fetching openshift-install binary
+
+**Risk**: openshift-install binary not available for required version
+- **Mitigation**: Leverage existing installer cache infrastructure with proper error handling
+
+**Risk**: Temporary directory cleanup failures could leak disk space
+- **Mitigation**: Implement proper cleanup with deferred functions and log warnings on cleanup errors
+
+## Design Details
+
+### API Changes
+
+**InfraEnv Image Type Addition:**
+Add new image type `disconnected-iso` to the InfraEnv model. This type indicates that the InfraEnv is intended for use with OVE ISOs.
+
+**Discovery.ign Endpoint Behavior:**
+The existing `/v2/infra-envs/{infra_env_id}/downloads/files?file_name=discovery.ign` endpoint behavior is modified:
+- For `disconnected-iso` type: Returns OVE ignition generated via openshift-install
+- For other types: Returns standard discovery ignition (unchanged)
+
+No new endpoints are added, maintaining API compatibility.
+
+### Temporary Cluster Management
+
+**Cluster Status Addition:**
+A new cluster status `disconnected` is added to identify temporary clusters used for OVE ISO configuration. This status:
+- Is set through the cluster state machine when a `disconnected-iso` type InfraEnv is registered for a cluster in `insufficient` status
```

Comment:
I still think we should also give this cluster a new kind.

We have ones for regular and imported, this can be a third. That would also give the UI a good way to filter them even when they are first created (before the transition to this special state is able to run)